src/main/java/org/mitre/inferno/Validator.java

```diff
@@ -132,30 +126,30 @@ public void loadProfile(byte[] profile) throws IOException {
     hl7Validator.getContext().cacheResource(resource);
   }
 
-  private IgResponse getLoadedIg(String id, String version) throws IOException {
-    NpmPackage npm = loadedPackages.get(id + "#" + version);
-    if (npm == null) {
-      npm = packageManager.loadPackage(id, version);
+  /**
+   * Finds any custom package that fits the given id and (possibly null) version.
+   *
+   * @param id the ID of the custom package
+   * @param version the version of the custom package, or null to return the first match
+   * @return a matching custom IG package, or null if no matching package was found
+   */
+  private NpmPackage findCustomPackage(String id, String version) {
+    String idRegex = "^" + id + "#" + (version != null ? version : ".*") + "$";
+    for (Map.Entry<String, NpmPackage> entry : loadedPackages.entrySet()) {
+      if (entry.getKey().matches(idRegex)) {
+        return entry.getValue();
+      }
     }
-    return getLoadedIg(npm);
+    return null;
   }
 
-  private IgResponse getLoadedIg(NpmPackage npm) throws IOException {
-    InputStream in = npm.load(".index.json");
-    JsonObject index = (JsonObject) JsonParser.parseString(TextFile.streamToString(in));
-
-    JsonArray files = index.getAsJsonArray("files");
-    List<String> profileUrls = new ArrayList<>();
-    for (JsonElement f : files) {
-      JsonObject file = (JsonObject) f;
-      String type = JSONUtil.str(file, "resourceType");
-      String url = JSONUtil.str(file, "url");
-      if (type.equals("StructureDefinition")) {
-        profileUrls.add(url);
-      }
+  private IgResponse getIg(String id, String version) throws IOException {
+    NpmPackage npm = findCustomPackage(id, version);
+    // Fallback to packages from packages.fhir.org if no custom packages match
+    if (npm == null) {
+      npm = packageManager.loadPackage(id, version);
     }
-    Collections.sort(profileUrls);
-    return new IgResponse(npm.id(), npm.version(), profileUrls);
+    return IgResponse.fromPackage(npm);
   }
```

Comment:
I tried combining `getIg` with `loadIg` because there's only a one-line difference, but I couldn't think of a good name for the combined method. Maybe `getOrFetch(..., boolean fetch)`? I'm open to suggestions

Reply:
Personally I think you could call it `getIg` safely, with `boolean fetchIfNotFound` or something like that.